### PR TITLE
*: rename and add tidb_wait_split_region_finish_backoff session var to set wait split region backoff time.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
* rename functions and variables.
* add a session variable `tidb_wait_split_region_finish_backoff` to set the split region back off time.
* change `tidb_wait_split_region_finish` default value to true.  We meet some performence problem cause by default `async scatter`  behaviour . So we decide change this default behaviour to `sync scatter`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: util/backoff/backoff_time.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package backoff
```
Comment (zimulala) [inline util/backoff/backoff_time.go]: Please add license.

Comment (crazycs520) [reply util/backoff/backoff_time.go]: done

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -379,6 +380,9 @@ type SessionVars struct {
 
 	// LowResolutionTSO is used for reading data with low resolution TSO which is updated once every two seconds.
 	LowResolutionTSO bool
+
+	// Backoff is used for set the back off time.
+	Backoff backoff.Time
```
Comment (zimulala) [inline sessionctx/variable/session.go]: We'd better rename this session variable. In TiDB we have a lot of `backoff`.

File: sessionctx/variable/session.go [hunk 3]
```diff
@@ -792,8 +797,10 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		s.SlowQueryFile = val
 	case TiDBEnableFastAnalyze:
 		s.EnableFastAnalyze = TiDBOptOn(val)
-	case TiDBWaitTableSplitFinish:
-		s.WaitTableSplitFinish = TiDBOptOn(val)
+	case TiDBWaitSplitRegionFinish:
+		s.WaitSplitRegionFinish = TiDBOptOn(val)
+	case TiDBWaitSplitRegionFinishBackoff:
+		s.Backoff.SetWaitScatterRegionFinishBackoff(int(tidbOptInt64(val, backoff.WaitScatterRegionFinishBackoff)))
```
Comment (zimulala) [inline sessionctx/variable/session.go]: I think we add function like `tidbOptPositiveInt32` is enough.  Then we remove `backoff_time.go`

File: sessionctx/variable/session.go (hunk 2 above)
Comment (crazycs520) [reply sessionctx/variable/session.go]: How about `BackoffVar` to mean this is a variable backoff time?

File: sessionctx/variable/session.go (hunk 3 above)
Comment (crazycs520) [reply sessionctx/variable/session.go]: done.

File: sessionctx/variable/session.go (hunk 2 above)
Comment (crazycs520) [reply sessionctx/variable/session.go]: done.

File: sessionctx/variable/session.go [hunk 4]
```diff
@@ -998,6 +997,25 @@ const (
...
+// GetWaitScatterRegionFinishBackoff gets the back off time of waitScatterRegionFinish.
+func (b *BackoffTimer) GetWaitScatterRegionFinishBackoff() int {
+	if b.waitScatterRegionFinish > 0 {
+		return b.waitScatterRegionFinish
+	}
+	return DefWaitScatterRegionFinishBackoff
+}
+
+// SetWaitScatterRegionFinishBackoff sets the back off time of waitScatterRegionFinish.
+func (b *BackoffTimer) SetWaitScatterRegionFinishBackoff(t int) {
```
Comment (zimulala) [inline sessionctx/variable/session.go]: I think we needn't this function. `tidbOptPositiveInt32` is do it.

Comment (crazycs520) [reply sessionctx/variable/session.go]: done. Thanks.

File: sessionctx/variable/session.go [hunk 5]
```diff
@@ -982,6 +988,19 @@ const (
...
 )
 
+// BackoffTimer contains variable back off time.
+type BackoffTimer struct {
+	waitScatterRegionFinish int
+}
+
+// GetWaitScatterRegionFinishBackoff gets the back off time of waitScatterRegionFinish.
+func (b *BackoffTimer) GetWaitScatterRegionFinishBackoff() int {
+	if b.waitScatterRegionFinish > 0 {
```
Comment (zimulala) [inline sessionctx/variable/session.go]: If we use `tidbOptPositiveInt32` to set the value. I think we needn't do this.

Comment (crazycs520) [reply sessionctx/variable/session.go]: done. Thanks.

**Pull request merged**